### PR TITLE
[KYUUBI #5282] Support configure Trino session conf in `kyuubi-default.conf`

### PR DESCRIPTION
**Pull request by ASiegeLion**
### _Why are the changes needed?_

Refer to spark and flink settings conf,  support configure Trino session conf in kyuubi-default.conf
issue : https://github.com/apache/kyuubi/issues/5282

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [ ] [Run test](https://kyuubi.readthedocs.io/en/master/contributing/code/testing.html#running-tests) locally before make a pull request

### _Was this patch authored or co-authored using generative AI tooling?_

No

### CODE REVIEW

**Comment by pan3793** [comment]
The idea looks fine, please update the PR description and docs to mention this feature.

**Comment by ASiegeLion** [comment]
> The idea looks fine, please update the PR description and docs to mention this feature.

ok，

**Comment by ASiegeLion** [comment]
> The idea looks fine, please update the PR description and docs to mention this feature.

docs updated

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala [hunk 1]
```diff
@@ -170,6 +174,12 @@ class TrinoSessionImpl(
     resultSet.next().head.toString
   }
 
+  private def getTrinoSessionConf(sessionConf: KyuubiConf): Map[String, String] = {
+    val trinoSessionConf = sessionConf.getAll.filterKeys(_.startsWith("trino."))
+      .map { case (k, v) => (k.stripPrefix("trino."), v) }
+    trinoSessionConf
```
Comment (cxzl25) [inline externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:180]:
```suggestion
    trinoSessionConf.toMap
```

Comment (cxzl25) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala:180]: scala 2.13
```
TrinoSessionImpl.scala:180: type mismatch;
 found   : scala.collection.View[(String, String)]
 required: Map[String,String]
```

File: docs/configuration/settings.md [hunk 2]
```diff
@@ -526,6 +526,34 @@ Setting them in the JDBC Connection URL supplies session-specific for each SQL e
 
 Please refer to the Flink official online documentation for [SET Statements](https://nightlies.apache.org/flink/flink-docs-stable/docs/dev/table/sql/set/)
 
+## Trino Configurations
```
Comment (cxzl25) [inline docs/configuration/settings.md:529]: Run `./dev/reformat`

Comment (ASiegeLion) [reply docs/configuration/settings.md:529]: ok

File: externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala (hunk 1 above)
Comment (ASiegeLion) [reply externals/kyuubi-trino-engine/src/main/scala/org/apache/kyuubi/engine/trino/session/TrinoSessionImpl.scala]: ok

**Comment by ASiegeLion** [comment]
sorry, Why the some check failed? @cxzl25

**Comment by pan3793** [comment]
The `settings.md` is auto-generated, I think you should update  `org.apache.kyuubi.config.AllKyuubiConfiguration` instead

**Comment by cxzl25** [comment]
Because you modified `docs/configuration/settings.md` directly, but this file is dynamically generated through `AllKyuubiConfiguration`, so the check fails.

**Comment by ASiegeLion** [comment]
> The `settings.md` is auto-generated, I think you should update `org.apache.kyuubi.config.AllKyuubiConfiguration` instead

thanks

**Comment by ASiegeLion** [comment]
> Because you modified `docs/configuration/settings.md` directly, but this file is dynamically generated through `AllKyuubiConfiguration`, so the check fails.

thanks

**Review by cxzl25 — APPROVED** [review]

File: docs/configuration/settings.md [hunk 3]
```diff
@@ -526,6 +526,33 @@ Setting them in the JDBC Connection URL supplies session-specific for each SQL e
...
+
+### Via  config.properties
```
Comment (pan3793) [inline docs/configuration/settings.md:531]:
```suggestion
### Via config.properties
```

File: kyuubi-server/src/test/scala/org/apache/kyuubi/config/AllKyuubiConfiguration.scala [hunk 4]
```diff
@@ -177,6 +177,36 @@ class AllKyuubiConfiguration extends KyuubiFunSuite {
       """Please refer to the Flink official online documentation for [SET Statements]
         |(https://nightlies.apache.org/flink/flink-docs-stable/docs/dev/table/sql/set/)"""
 
+    builder ++=
+      """## Trino Configurations
+        |### Via  config.properties""" += """Setting them in `$TRINO_HOME/etc/config.properties`
```
Comment (pan3793) [inline kyuubi-server/src/test/scala/org/apache/kyuubi/config/AllKyuubiConfiguration.scala:182]:
```suggestion
        |### Via config.properties""" += """Setting them in `$TRINO_HOME/etc/config.properties`
```

**Review by pan3793 — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Comment by pan3793** [comment]
Merged to master/1.8

**Pull request closed without merging**